src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -13,42 +13,152 @@ import { IStorageService, StorageScope } from 'vs/platform/storage/common/storag
+		const monthlyLastUseDate: number  = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const firstTimeUser: boolean = dailyLastUseDate === Date.parse('0');
 
-		let today = new Date().toUTCString();
+		let todayString: string = new Date().toUTCString();
 
 		// daily user event
-		if (this.diffInDays(Date.parse(today), dailyLastUseDate) >= 1) {
+		if (this.didDayChange(dailyLastUseDate)) {
 			// daily first use
 			telemetryService.publicLog('telemetry.dailyFirstUse', { dailyFirstUse: true });
-			storageService.store('telemetry.dailyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.dailyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
 		// weekly user event
-		if (this.diffInDays(Date.parse(today), weeklyLastUseDate) >= 7) {
+		if (this.didWeekChange(weeklyLastUseDate)) {
 			// weekly first use
 			telemetryService.publicLog('telemetry.weeklyFirstUse', { weeklyFirstUse: true });
-			storageService.store('telemetry.weeklyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.weeklyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
-		// monthly user events
-		if (this.diffInDays(Date.parse(today), monthlyLastUseDate) >= 30) {
+
+		/* send monthly uses at the end of month */
+		const monthlyUseCount: number = storageService.getNumber('telemetry.monthlyUseCount', StorageScope.GLOBAL, 0);
+		if (this.didMonthChange(monthlyLastUseDate)) {
 			telemetryService.publicLog('telemetry.monthlyUse', { monthlyFirstUse: true });
-			storageService.store('telemetry.monthlyLastUseDate', today, StorageScope.GLOBAL);
+			// the month changed, so send the user usage type event based on monthly count for last month
+			// and reset the count for this month
+			let lastMonthDate = new Date(monthlyLastUseDate);
+			this.sendUsageEvent(monthlyUseCount, lastMonthDate);
+
+			const wasActiveLastMonth: boolean = storageService.getBoolean('telemetry.wasActiveLastMonth', StorageScope.GLOBAL, false);
+
+			if (firstTimeUser) {
+				// new user
+				this.sendGrowthTypeEvent(UserGrowthType.NewUser, lastMonthDate);
+			}
+
+			// continuing or returning user
+			this.sendGrowthTypeEvent(wasActiveLastMonth ? UserGrowthType.ContinuingUser : UserGrowthType.ReturningUser, lastMonthDate);
+
+			// set wasActiveUserLastMonth
+			storageService.store('telemetry.wasActiveLastMonth', true, StorageScope.GLOBAL);
+
+			// reset the monthly count for the new month
+			storageService.store('telemetry.monthlyUseCount', 1, StorageScope.GLOBAL);
+			storageService.store('telemetry.monthlyLastUseDate', todayString, StorageScope.GLOBAL);
+		} else {
+			// if it's the same month, increment the monthly use count
+			storageService.store('telemetry.monthlyUseCount', monthlyUseCount+1, StorageScope.GLOBAL);
+		}
+	}
+
+	private didDayChange(lastUseDateNumber: number): boolean {
```

Comment:
For now this is fine - but let's keep these in mind since we might end up using the same logic elsewhere (in which case we should move it into a more generic location)